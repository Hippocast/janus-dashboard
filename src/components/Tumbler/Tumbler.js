import React from 'react';
import PropTypes from 'prop-types';

import { Field } from 'redux-form';

import block from '../../helpers/bem-cn';

import './Tumbler.css';

const b = block('j-tumbler');

const propTypes = {
    name: PropTypes.string.isRequired,
};

const Tumbler = ({ name }) => {
    return (
        <div className={b()}>
            <Field
                name={name}
                className={b('checkbox')()}
                component="input"
                type="checkbox"
                normalize={v => !!v}
            />
            <div className={b('icon')()}></div>
        </div>
    );
};

Tumbler.propTypes = propTypes;

export default Tumbler;
