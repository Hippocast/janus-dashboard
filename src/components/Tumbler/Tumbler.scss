@import 'index.scss';

$size: 4px;
$height: $size*4;
$width: $size*7;
$circleSize: $size*4;

.j-tumbler {
    position: relative;
    width: $width;

    &__icon {
        position: relative;
        z-index: 0;
        height: $height;

        &:before,
        &:after {
            @include pseudo();
            box-sizing: border-box;
        }

        &:before {
            width: $width;
            height: $height;
            background-color: $hf-color-gray-pale;
            border-radius: $height;
            border: solid 1px darken($hf-color-gray-pale, 7%);
            top: 0;
            z-index: 1;
            transition: background-color $hf-time, border-color $hf-time;
        }

        &:after {
            background-color: $hf-color-white;
            width: $circleSize;
            height: $circleSize;
            border-radius: 50%;
            top: 0;
            left: 0;
            z-index: 2;
            box-shadow: $hf-box-shadow-tumbler;
            transform: translateX(0);
            transition: all .15s;
        }
    }

    &__checkbox {
        position: absolute;
        height: 100%;
        width: 100%;
        left: 0;
        padding: 0;
        margin: 0;
        box-sizing: border-box;
        opacity: 0;
        pointer-events: all;
        z-index: 2;
        cursor: pointer;

        &:checked + .j-tumbler__icon {
            &:before {
                background-color: $hellofresh-color;
                border: solid 1px darken($hellofresh-color, 7%);
            }
            &:after {
                left: $width;
                transform: translateX(-100%);
            }
        }
    }
}
