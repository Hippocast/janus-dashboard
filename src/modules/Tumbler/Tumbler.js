import React from 'react';
import { Field } from 'redux-form';

import block from '../../helpers/bem-cn';

import './Tumbler.css';

const b = block('j-tumbler');

const Tumbler = ({ name }) => {
    return (
        <div className={b()}>
            <Field
                name={name}
                className={b('checkbox')}
                component="input"
                type="checkbox"
                normalize={v => !!v}
            />
            <div className={b('icon')}></div>
        </div>
    );
};

export default Tumbler;
