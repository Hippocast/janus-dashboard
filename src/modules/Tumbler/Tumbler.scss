@import 'index.scss';

.j-tumbler {}
