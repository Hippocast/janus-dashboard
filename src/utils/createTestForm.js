import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';

const Form = ({ children, initialValues }) => children;

const createTestForm = initialValues => connect(
    () => ({
        initialValues
    }),
    null,
)(
    reduxForm({
        form: 'mockForm',
    })(Form)
);

export default createTestForm;
