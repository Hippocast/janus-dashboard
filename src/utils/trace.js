import R from 'ramda';
import { openResponseModal } from '../store/actions';

/**
 * @name trace
 * @desc helper function for development only.
 *          It should be used like this:
 * @example (from https://github.com/MostlyAdequate/mostly-adequate-guide/blob/master/ch5.md#debugging)
 *
 * import trace from '../utilz/trace';
 *
 * const dasherize = R.compose(
 *      join('-'),
 *      toLower,
 *      trace('after split'),
 *      split(' '),
 *      replace(/\s{2,}/ig, ' ')
 * );
 *
 * dasherize('The world is a vampire');
 * // after split [ 'The', 'world', 'is', 'a', 'vampire' ]
 */
const trace = R.curry((tag, x) => {
    console.error('trace -> ', tag, x);
    return x;
});

export default trace;
