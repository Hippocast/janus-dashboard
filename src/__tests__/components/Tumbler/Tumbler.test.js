import React from 'react';
import { createStore } from 'redux';

import { wrap } from '../../../utils/createTestForm';
import Tumbler from '../../../components/Tumbler/Tumbler';

const initialValues = {
    'mock-name': '',
};

const store = createStore(() => ({
    form: {
        mockForm: {
            values: initialValues,
        }
    }
}));

describe('Tumbler component', () => {
    const requiredProps = {
        name: 'mock-name',
    };

    it('renders correctly', () => {
        const tree = wrap(store)(initialValues)(
            <Tumbler
                {...requiredProps}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
