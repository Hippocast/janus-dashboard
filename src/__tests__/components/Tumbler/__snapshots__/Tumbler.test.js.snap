// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tumbler component renders correctly 1`] = `
<div
  className="j-tumbler"
>
  <input
    checked={false}
    className={[Function]}
    name="mock-name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value=""
  />
  <div
    className={[Function]}
  />
</div>
`;
